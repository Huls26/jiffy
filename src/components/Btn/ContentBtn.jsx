/* eslint-disable react/require-default-props */
import PropTypes from 'prop-types';
import contenBtnStyle from './styles/contentBtnStyle';

export default function ContentBtn({
  text, bg, onClick, type,
}) {
  const bgCol = bg || 'bg-yellow-1';

  return (
    <button
      onClick={onClick}
      type={!type ? 'button' : 'submit'}
      className={contenBtnStyle(bgCol).btnStyle}
    >
      {text}
    </button>
  );
}

ContentBtn.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  text: PropTypes.any.isRequired,
  bg: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};
