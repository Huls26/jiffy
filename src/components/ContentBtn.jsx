import PropTypes from 'prop-types';

export default function ContentBtn({ text, bg }) {
  const bgColor = bg || 'bg-yellow-1';
  return (
    <button
      type="button"
      className={`
        ${bgColor} px-3 py-1
        font-A font-bold text-sm 
        grow-0 basis-0
        border-dark-2
        border border-b-2 border-r-2 rounded
        capitalize
        hover:opacity-80
        active:bg-green
      `}
    >
      {text}
    </button>
  );
}

ContentBtn.propTypes = {
  text: PropTypes.string.isRequired,
  bg: PropTypes.string.isRequired,
};
