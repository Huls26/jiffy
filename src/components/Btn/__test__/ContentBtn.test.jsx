import {
  describe, it, expect, vi,
} from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import ContentBtn from '../ContentBtn';

describe('ContentBtn', () => {
  it('renders correctly', () => {
    const { container } = render(<ContentBtn text="Submit" />);

    expect(container).toMatchSnapshot();
  });

  it('calls the onClick prop when clicked', () => {
    // eslint-disable-next-line no-undef
    const onClick = vi.fn();
    const { container } = render(<ContentBtn text="Submit" onClick={onClick} />);

    fireEvent.click(container.querySelector('button'));

    expect(onClick).toHaveBeenCalled();
  });
});
