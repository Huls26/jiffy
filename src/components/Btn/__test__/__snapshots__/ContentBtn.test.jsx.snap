// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentBtn > renders correctly 1`] = `
<div>
  <button
    class="
        bg-yellow-1 px-3 py-1
        font-A font-bold text-sm 
        grow-0 basis-0
        border-dark-2
        border border-b-2 border-r-2 rounded
        capitalize
        hover:opacity-80
        active:bg-green
      "
    type="button"
  >
    Submit
  </button>
</div>
`;
